File: src/Tests/Behavioral/TypeConversion/TypeConversion.cs
using go;

namespace go;

using fmt = fmt_package;

partial class main_package {

private static void Main() {
    [GoType] partial struct Person {
        public @string Name;
        public ж<struct{Street string; City string}> Address;
    }

    ж<struct{Name string "json:\"name\""; Address *struct{Street string "json:\"street\""; City string "json:\"city\""} "json:\"address\""}> data = default!;
    ref var mine = ref heap(new Person(), out var Ꮡmine);
    ж<Person> person = (Person.val)(data);
    person = Ꮡmine;
    fmt.Println(mine == person.val);
    fmt.Println(slice<rune>(((@string)"白鵬翔"u8)));
}

} // end main_package

File: src/Tests/Behavioral/TypeConversion/TypeConversion.go
package main

import "fmt"

type Person struct {
	Name    string
	Address *struct {
		Street string
		City   string
	}
}

func main() {
	var data *struct {
		Name    string `json:"name"`
		Address *struct {
			Street string `json:"street"`
			City   string `json:"city"`
		} `json:"address"`
	}
	
	var mine Person

	var person = (*Person)(data)  // ignoring tags, the underlying types are identical
	
	person = &mine
	
	fmt.Println(mine == *person)
    
    fmt.Println([]rune(string("白鵬翔"))) 
}
